draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -321,26 +321,46 @@ q=foo&amp;limit=10&amp;sort=-published
 
   <section title="The &quot;Accept-Query&quot; Header Field" anchor="field.accept-query">
     <t>
-      The "Accept-Query" response header field &MAY; be used by a resource to
+      The "Accept-Query" response header field can be used by a resource to
       directly signal support for the QUERY method while identifying
       the specific query format media type(s) that may be used.
     </t>
-<sourcecode type="abnf">
-Accept-Query = 1#media-type
-</sourcecode>
     <t>
-      The Accept-Query header field specifies a comma-separated listing of media
-      types (with optional parameters) as defined by
-      <xref target="HTTP" section="8.3.1"/>. <cref>field syntax currently discussed in <eref target="https://github.com/httpwg/http-extensions/issues/2860"/></cref>
+      "Accept-Query" contains a list of media ranges (<xref target="HTTP" section="12.5.1"/>)
+      using "Structured Fields" syntax (<xref target="STRUCTURED-FIELDS"/>).
+      Media ranges are represented by a List Structured Header Field of either Tokens or
+      Strings, containing the media range value without parameters. Parameters,
```

Comment:
saying "Parameters, if any, are mapped to Parameters of type String" is both unnecessarily prescriptive and confusing. if i say `application/foo+json; x=3.1`, where parameter `x` is (or could be interpreted as) a SF decimal, i should (1) be allowed to say it that way (that is, i shouldn't be **forced** to say `application/foo+json; x="3.1"`), and (2) it's nobody else's business whether i, when receiving this field, treat the `3.1` as a string or a number. that should be between the definition & semantics of the media-type/range and my implementation.

how about words like
> "Parameters of the media-range, if any, are mapped to Structured Field Parameters. The media-range and its parameters \<bcp14>MUST\</bcp14> be parsable as a valid Structured Field."

perhaps using "expressed" or "serialized" in place of "mapped" in the first sentence.

Reply:
The restriction is intentional. For instance, when not quoting values, surprising things may happen (integer vs floats etc). 

On-the-wire verbosity really is no issue here.

Reply:
media types & ranges have an established syntactic form & semantics, and an established ecosystem beyond HTTP. we want to use this field to tell clients what media types (or ranges of them) are acceptable for this method.

> The restriction is intentional. For instance, when not quoting values, surprising things may happen (integer vs floats etc).

i don't think this restriction is a _necessary_ one. i'm not convinced that guarding against a hypothetical insufficient implementation warrants restricting otherwise valid expressions of media types/ranges in a structured field. i think the only restrictions should be the minimum actually necessary so that an expression of a media type/range is **syntactically** valid as both a `media-range` and a structured field, thereby maintaining as much compatibility as possible with the media-type ecosystem (and all of the different normative and common forms of different media types) while conforming to Structured Field syntax.